### PR TITLE
Pp 3530 smartpay 3ds response connector

### PR DESCRIPTION
**Pull request by georgeracu**
## WHAT
Handle 3ds Response from SmartPay on `connector`

## HOW 
- Augmented Auth3dsDetails with the new parameter for 3DS SmartPay
- Augmented SmartPay OrderRequestBuilder and PaymentProvider with
the new 3DS params and enabled 3DS Response
- Added xml template for 3ds response

with @chrisgrimble

### CODE REVIEW

**Review by daniele-occhipinti — APPROVED** [review]
LGTM.

File: src/main/java/uk/gov/pay/connector/service/smartpay/SmartpayPaymentProvider.java [hunk 1]
```diff
@@ -146,7 +145,7 @@ public ExternalChargeRefundAvailability getExternalChargeRefundAvailability(Char
     private Function<AuthorisationGatewayRequest, GatewayOrder> buildAuthoriseOrderFor() {
         return request -> {
             SmartpayOrderRequestBuilder smartpayOrderRequestBuilder = request.getGatewayAccount().isRequires3ds() ?
-                    aSmartpayAuthorise3dsOrderRequestBuilder() : aSmartpayAuthoriseOrderRequestBuilder();
+                    aSmartpay3dsRequiredOrderRequestBuilder() : aSmartpayAuthoriseOrderRequestBuilder();
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/connector/service/smartpay/SmartpayPaymentProvider.java:148]: 👍

File: src/main/resources/templates/smartpay/Smartpay3dsRequiredOrderTemplate.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                <ns2:city>${authCardDetails.address.city?xml}</ns2:city>
+                <ns2:country>${authCardDetails.address.country?xml}</ns2:country>
+            </ns1:billingAddress>
+        </ns1:card>
+        <ns1:merchantAccount>${merchantCode}</ns1:merchantAccount>
+        <ns1:reference>${paymentPlatformReference?xml}</ns1:reference>
+        <ns1:shopperReference>${description?xml}</ns1:shopperReference>
+    </ns1:paymentRequest>
+</ns1:authorise>
+        </soap:Body>
```
Comment (kakumara) [inline src/main/resources/templates/smartpay/Smartpay3dsRequiredOrderTemplate.xml:49]: minor: Not aligned properly 😑

**Review by kakumara — COMMENTED** [review]
Its probably good to have a contract test also for authorise 3d call. We can hardcode a existing `paResponse` and a `md` on smartpay so that we can simply run it by changing it whenever required. (Even though it will be ignored in codebase)

**Pull request merged**